### PR TITLE
ENT-2747/master: Tighten Postgresql access permissions

### PR DESCRIPTION
**Pull request by nickanderson**
- Removed executable bit from unix socket
- Removed local TCP access rules

Ticket: ENT-2747
Changelog: Commit

Merge together:
https://github.com/cfengine/buildscripts/pull/668
cfengine/nova#1585
cfengine/mission-portal#999

### CODE REVIEW

**Comment by nickanderson** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4456/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4456/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4456/

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4458/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4458/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4458/

File: deps-packaging/postgresql/debian/rules [hunk 1]
```diff
@@ -28,9 +28,11 @@ install: build
 	make -C contrib install DESTDIR=$(CURDIR)/debian/tmp
 	rm -f $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/postgresl.conf.cfengine
 	patch -d $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/ -o postgresql.conf.cfengine < $(CURDIR)/postgresql.conf.cfengine.patch
+	patch -d $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/ -o postgresql.conf.sample < $(CURDIR)/postgresql.conf.sample.patch
 	chmod --reference $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/postgresql.conf.sample $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/postgresql.conf.cfengine
 	chown --reference $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/postgresql.conf.sample $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/postgresql.conf.cfengine
-	rm -f $(CURDIR)/debian/tmp$(PREFIX)/lib/*.a
+	patch -d $(CURDIR)/debian/tmp$(PREFIX)/share/postgresql/ -o pg_hba.conf.sample < $(CURDIR)/pg_hba.conf.socket-only.patch
+  rm -f $(CURDIR)/debian/tmp$(PREFIX)/lib/*.a
```
Comment (vpodzime) [inline deps-packaging/postgresql/debian/rules]: This has to be a TAB.

**Comment by aleksandrychev** [comment]
Merge together:
https://github.com/cfengine/nova/pull/1585
https://github.com/cfengine/mission-portal/pull/999

File: deps-packaging/postgresql/debian/rules (hunk 1 above)
Comment (nickanderson) [reply deps-packaging/postgresql/debian/rules]: F TABS

Comment (nickanderson) [reply deps-packaging/postgresql/debian/rules]: Changed, pushed.

**Comment by vpodzime** [comment]
Please squash the commits.

**Comment by aleksandrychev** [comment]
@vpodzime Jenkins still shows me an error
```dpkg-buildpackage: error: fakeroot debian/rules binary gave error exit status 2``` https://ci.cfengine.com/job/testing-pr/6519/label=PACKAGES_HUB_x86_64_linux_ubuntu_16/console

**Comment by vpodzime** [comment]
Yes, because one of the patch files doesn't apply:
```
11:32:08 patching file postgresql.conf.sample
11:32:08 Hunk #1 FAILED at 65.
11:32:08 1 out of 1 hunk FAILED -- saving rejects to file postgresql.conf.sample.rej
```

File: deps-packaging/postgresql/postgresql.conf.sample.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+--- postgresql.conf.sample	2019-11-11 16:03:10.000000000 -0600
++++ postgresql.conf.sample.cfengine	2020-01-21 13:07:32.232910689 -0600
```
Comment (aleksandrychev) [inline deps-packaging/postgresql/postgresql.conf.sample.patch]: please replace `postgresql.conf.sample.cfengine` with `postgresql.conf.sample` to fix a [patch issue](https://github.com/cfengine/buildscripts/pull/668#issuecomment-578140408)

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4494/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4494/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4494/

**Comment by aleksandrychev** [comment]
well, the issue is still there, I will test again, but patch worked locally...
image

**Comment by aleksandrychev** [comment]
I tried a direct path to the file `patch /var/cfengine/share/postgresql/postgresql.conf.sample < /vagrant/postgresql.conf.sample.patch`  without arguments and this works. We use `-d /var/cfengine/share/postgresql/ -o postgresql.conf.sample` and when I removed  `-o postgresql.conf.sample` also patch works properly. I don't know why...

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4497/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4497/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4497/

**Comment by nickanderson** [comment]
@cf-bottom  jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4499/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4499/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4499/

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4540/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4540/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4540/

**Comment by vpodzime** [comment]
@cf-bottom jenkins again, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4544/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4544/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4544/

**Comment by vpodzime** [comment]
> Alright, I triggered a build:
> 
> [Build Status](https://ci.cfengine.com//job/pr-pipeline/4544/)
> 
> **Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4544/

Weird failure on Ubuntu 16. I retriggered the build:
[Build Status](https://ci.cfengine.com/job/pr-pipeline/4545/)

**Comment by nickanderson** [comment]
Waiting on https://github.com/cfengine/nova/pull/1585

**Pull request merged**